bftengine/tests/testSeqNumForClientRequest/seqNumForClientRequest_test.cpp

```diff
@@ -1,8 +1,80 @@
 #include "gtest/gtest.h"
+#include "SimpleClientImp.cpp"
 
-TEST(seqNumForClientRequest_test, initial_test) {}
+using namespace std;
+using namespace bftEngine;
 
-int main(int argc, char* argv[]) {
+namespace {
+
+class mockSeqNumGenerator : public SeqNumberGeneratorForClientRequestsImp {
+ public:
+  uint32_t getLastCountOfUniqueFetchID() { return lastCountOfUniqueFetchID_; }
+
+  void setLastCountOfUniqueFetchID(uint64_t count) { lastCountOfUniqueFetchID_ = count; }
+};
+
+TEST(seqNumForClientRequest_test, send_request_fast_still_persistence) {
+  SeqNumberGeneratorForClientRequests *seqNumGen = new mockSeqNumGenerator();
+
+  set<u_int64_t> seqNums;
+
+  for (int i = 0; i < 10; i++) {
+    seqNums.insert(seqNumGen->generateUniqueSequenceNumberForRequest());
+  }
+
+  Assert(seqNums.size() == 10);
+
+  delete seqNumGen;
+}
+
+TEST(seqNumForClientRequest_test, reach_counter_max_value) {
+  SeqNumberGeneratorForClientRequests *seqNumGen = new mockSeqNumGenerator();
+
+  set<u_int64_t> seqNums;
+
+  std::chrono::time_point<std::chrono::system_clock> now = std::chrono::system_clock::now();
+
+  seqNums.insert(seqNumGen->generateUniqueSequenceNumberForRequest(now));
+
+  auto mockSeqNumGen = dynamic_cast<mockSeqNumGenerator *>(seqNumGen);
+
+  // Set lastCount biggest value.
+  mockSeqNumGen->setLastCountOfUniqueFetchID(0x3FFFFF);
+
+  seqNums.insert(seqNumGen->generateUniqueSequenceNumberForRequest(now));
+
+  ASSERT_EQ(seqNums.size(), 2);
```

Comment:
shouldn't we also check the content of seqNums? For example, that they are always increasing, or other, more advanced properties?

Reply:
The content here is a number generated by a unique time stamp, I mostly tested the uniqueness this class trying to achieve.

Reply:
Aren't the generated sequence numbers also supposed to be monotonically increasing? If so, let's also test this.